File: app/src/main/java/com/tasomaniac/devwidget/LifecycleScopeModule.java
package com.tasomaniac.devwidget;

import com.uber.autodispose.android.lifecycle.AndroidLifecycleScopeProvider;

import androidx.fragment.app.FragmentActivity;
import dagger.Module;
import dagger.Provides;

@Module
public abstract class LifecycleScopeModule {

    @Provides
    static AndroidLifecycleScopeProvider lifecycleScopeProvider(FragmentActivity activity) {
        return AndroidLifecycleScopeProvider.from(activity);
    }
}

File: base/src/main/kotlin/com/tasomaniac/devwidget/LifecycleScopeModule.kt
package com.tasomaniac.devwidget

import androidx.fragment.app.FragmentActivity
import com.uber.autodispose.ScopeProvider
import com.uber.autodispose.android.lifecycle.AndroidLifecycleScopeProvider
import dagger.Module
import dagger.Provides

@Module
object LifecycleScopeModule {

    @Provides
    @JvmStatic
    internal fun lifecycleScopeProvider(activity: FragmentActivity): ScopeProvider =
        AndroidLifecycleScopeProvider.from(activity)
}
